nautobot/extras/models/mixins.py

```diff
@@ -14,19 +14,101 @@
+            self._dynamic_group_uncached_queryset = queryset
+
+        return DynamicGroup.objects.get_for_object(self, skip_cache=True)
+
+    @property
+    def dynamic_groups_cached(self):
+        """
+        Return a queryset of `DynamicGroup` objects this instance is a member of.
+
+        This will use a cached member list of the dynamic groups if available.
+
+        In optimal conditions this will incur a single database query to convert internal list into a queryset which is reasonably performant.
+        """
+        from nautobot.extras.models.groups import DynamicGroup
+
+        if not hasattr(self, "_dynamic_group_cached_queryset"):
             queryset = DynamicGroup.objects.get_for_object(self)
-            self._dynamic_group_queryset = queryset
+            self._dynamic_group_cached_queryset = queryset
+
+        return self._dynamic_group_cached_queryset
+
+    @property
+    def dynamic_groups_list(self):
+        """
+        Return a list of `DynamicGroup` objects this instance is a member of.
+
+        This will NOT use a cached member list of the dynamic groups and will always query the database for each DynamicGroup.
 
-        return self._dynamic_group_queryset
+        This saves a final query to turn the list into a queryset.
+        """
+        from nautobot.extras.models.groups import DynamicGroup
+
+        if not hasattr(self, "_dynamic_group_uncached_list"):
+            dg_list = DynamicGroup.objects.get_list_for_object(self, skip_cache=True)
+            self._dynamic_group_uncached_list = dg_list
+
+        return self._dynamic_group_uncached_list
+
+    @property
+    def dynamic_groups_list_cached(self):
+        """
+        Return a list of `DynamicGroup` objects this instance is a member of.
+
+        This will use a cached member list of the dynamic groups if available.
+
+        In optimal conditions this will incur no database queries.
+        """
+
+        from nautobot.extras.models.groups import DynamicGroup
+
+        if not hasattr(self, "_dynamic_group_cached_list"):
+            dg_list = DynamicGroup.objects.get_list_for_object(self)
+            self._dynamic_group_cached_list = dg_list
+
+        return self._dynamic_group_cached_list
+
+    def get_dynamic_groups(self, as_queryset=False, skip_cache=False):
+        """
+        Return a list or queryset of `DynamicGroup` objects this instance is a member of.
+
+        Args:
+            as_queryset (bool): Return a queryset instead of a list. Optional, defaults to False.
+            skip_cache (bool): Skip the cache and query the database directly. Optional, defaults to False.
+        """
+        if as_queryset:
+            if skip_cache:
+                return self.dynamic_groups
+            else:
+                return self.dynamic_groups_cached
+        else:
+            if skip_cache:
+                return self.dynamic_groups_list
+            else:
+                return self.dynamic_groups_list_cached
```

Comment:
Same comment here as above - why have two different ways to do the same thing? Why would I want to call `.get_dynamic_groups(as_queryset=False, skip_cache=False)` instead of `.dynamic_groups_list_cached`?